src/main/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -292,7 +294,9 @@ public static LinearIndex createLinearIndex(final File inputFile, final FeatureC
     public static void writeIndex(final Index idx, final File idxFile) throws IOException {
         LittleEndianOutputStream stream = null;
         try {
-            stream = new LittleEndianOutputStream(new BufferedOutputStream(new FileOutputStream(idxFile)));
+            final OutputStream underlyingStream = (idx instanceof TabixIndex) ?
+                    new BlockCompressedOutputStream(idxFile) : new FileOutputStream(idxFile);
```

Comment:
I was thinking about this before, because actually it makes more sense to have a `Index.write(File idxFile)` method instead of the factory. But I decided to go for this to maintain compatibility. In addition, the javadoc said that "It is the responsibility of this class to determine and create the correct index type from the input file or stream.", so that's why I did it like this.